depot/containerstore/dependencymanager.go

```diff
@@ -76,6 +83,8 @@ func (bm *dependencyManager) DownloadCachedDependencies(logger lager.Logger, mou
 	for {
 		select {
 		case err := <-errChan:
+			close(cancelChan)
+			wg.Wait()
```

Comment:
Does this imply that if there's an error, we'll continue downloading the other dependencies? What if one of other downloads also has an error? Will `wg.Done()` get called?

Reply:
yeah wg.Done is in defer, so it should be called even if there is an error. You right we will wait for all go routines to finish, since we send to cancel channel they would be cancelled, even in the middle of io.Copy. This way we should not leak goroutines.

Reply:
I'm worried that we'll continue to leak downloaded files. The `bindMounts` collection that we return in the next line (along with the error that was received on the `errChan`) is only updated in the other case of this switch statement (`bindMounts.AddBindMount(cachedMount.CacheKey, cachedMount.BindMount)`). That suggests that other goroutines will continue their downloads, but not capture them in the `bindMounts` collection, and therefore be orphaned.

I suppose that's the behavior we already have, since the error case already doesn't account for any files that are downloaded by other goroutines. Just wanted to point out that we may still be orphaning files here.

Reply:
Ok, I should have read this change more closely. Because the errChan path is closing the `cancelChan` (which is passed to each of the download goroutines, the downloads will all get canceled. There's even a test for this!